src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -683,7 +732,7 @@ private static async Task<OrchestrationState> GetOrchestrationInstanceStateAsync
                 return;
             }
 
-            if (IsOrchestrationRunning(status))
+            if (IsOrchestrationRunning(status) || IsOrchestrationSuspended(status))
```

Comment:
I would recommend modifying the `IsOrchestrationRunning` method so that it's logic accounts for the Suspended state.

Reply:
Aren't they different, though? For instance, while here it makes sense to check for running or suspended, in other instances it makes sense to check just one or the other (see `SuspendAsync` and `TerminateAsync`).

Reply:
Let's discuss this one offline. I do see that you call this in `SuspendAsync` why we couldn't add the Suspended check in there.

Reply:
Fixed based on our conversation earlier. See https://github.com/Azure/azure-functions-durable-extension/pull/2227#discussion_r930447912